homeassistant/components/overkiz/switch.py

```diff
@@ -0,0 +1,138 @@
+"""Support for Overkiz switches."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import Any
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+from pyoverkiz.enums.ui import UIClass, UIWidget
+from pyoverkiz.types import StateType as OverkizStateType
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantOverkizData
+from .const import DOMAIN
+from .entity import OverkizDescriptiveEntity
+
+
+@dataclass
+class OverkizSwitchDescriptionMixin:
+    """Define an entity description mixin for switch entities."""
+
+    turn_on: Callable[[Callable[..., Awaitable[None]]], Awaitable[None]]
+    turn_off: Callable[[Callable[..., Awaitable[None]]], Awaitable[None]]
+    is_on: Callable[[Callable[[str], OverkizStateType]], bool]
+
+
+@dataclass
+class OverkizSwitchDescription(SwitchEntityDescription, OverkizSwitchDescriptionMixin):
+    """Class to describe an Overkiz switch."""
+
+
+SWITCH_DESCRIPTIONS: list[OverkizSwitchDescription] = [
+    OverkizSwitchDescription(
+        key=UIWidget.DOMESTIC_HOT_WATER_TANK,
+        turn_on=lambda execute_command: execute_command(
+            OverkizCommand.SET_FORCE_HEATING, OverkizCommandParam.ON
+        ),
+        turn_off=lambda execute_command: execute_command(
+            OverkizCommand.SET_FORCE_HEATING, OverkizCommandParam.OFF
+        ),
+        is_on=lambda select_state: (
+            select_state(OverkizState.IO_FORCE_HEATING) == OverkizCommandParam.ON
+        ),
+        icon="mdi:water-boiler",
+    ),
+    OverkizSwitchDescription(
+        key=UIClass.ON_OFF,
+        turn_on=lambda execute_command: execute_command(OverkizCommand.ON),
+        turn_off=lambda execute_command: execute_command(OverkizCommand.OFF),
+        is_on=lambda select_state: (
+            select_state(OverkizState.CORE_ON_OFF) == OverkizCommandParam.ON
+        ),
+        device_class=SwitchDeviceClass.OUTLET,
+    ),
+    OverkizSwitchDescription(
+        key=UIClass.SWIMMING_POOL,
+        turn_on=lambda execute_command: execute_command(OverkizCommand.ON),
+        turn_off=lambda execute_command: execute_command(OverkizCommand.OFF),
+        is_on=lambda select_state: (
+            select_state(OverkizState.CORE_ON_OFF) == OverkizCommandParam.ON
+        ),
+        icon="mdi:pool",
+    ),
+    OverkizSwitchDescription(
+        key=UIWidget.RTD_INDOOR_SIREN,
+        turn_on=lambda execute_command: execute_command(OverkizCommand.ON),
+        turn_off=lambda execute_command: execute_command(OverkizCommand.OFF),
+        is_on=lambda select_state: (
```

Comment:
If this entity is stateless, what does the `is_on` property mean?

Reply:
You are correct. I will double check this, but I think I accidentally copied this for a stateless device... Otherwise I would have mapped it to the siren entity.

Reply:
Good catch.